test/AccessController.ts

```diff
@@ -0,0 +1,95 @@
+        /* children with init need to add initializer modifier internally which will trigger this error */
+        await expect(initAsNonAdminPromise).to.be.revertedWith("Contract is not initialized");
+    });
+
+    it("Should allow admin to init", async function () {
+        await accessControllerMock.init(initArg);
+        const initialized = await accessControllerMock.getInitialized();
+        await expect(initialized).to.be.true;
+    });
+
+    it("Should execute code in child contract init function as part of AccessController.initializer", async function () {
+        /* someArg is set in mock child's init function: */
+        const someArg = await accessControllerMock.someArg();
+        expect(someArg).to.equal(initArg);
+    });
+
+    it("Should not allow admin to init twice", async function () {
+        const initPromise = accessControllerMock.init(1234);
+        await expect(initPromise).to.be.revertedWith("initializer: Already initialized");
+    });
+
+    it("Should revert if roles not set while getting role address", async function () {
+        const getAddressAdminPromise = accessControllerMock.getAddressAdmin();
+        const getAddressExecutivePromise = accessControllerMock.getAddressExecutive();
+        const getAddressGovernorPromise = accessControllerMock.getAddressGovernor();
+        const getAddressGuardianPromise = accessControllerMock.getAddressGuardian();
+        /* children with init need to add initializer modifier internally which will trigger this error */
+        await expect(getAddressAdminPromise).to.be.revertedWith("Roles have not been set up");
+        await expect(getAddressExecutivePromise).to.be.revertedWith("Roles have not been set up");
+        await expect(getAddressGovernorPromise).to.be.revertedWith("Roles have not been set up");
+        await expect(getAddressGuardianPromise).to.be.revertedWith("Roles have not been set up");
+    });
+
+    it("Should not allow non admin to call setRoles", async function () {
+        const setRolesPromise = accessControllerMock.connect(addr1).setRoles(owner.address, addr1.address, addr2.address);
+        await expect(setRolesPromise).to.be.revertedWith("Caller is not admin");
+    });
+
+    it("Should allow admin to call setRoles", async function () {
+        const executive = addr3.address;
+        const governor = addr1.address;
+        const guardian = addr2.address;
+        await accessControllerMock.setRoles(executive, governor, guardian);
+        const executiveSet = await accessControllerMock.getAddressExecutive();
+        const governorSet = await accessControllerMock.getAddressGovernor();
+        const guardianSet = await accessControllerMock.getAddressGuardian();
+        await expect(executive).to.equal(executiveSet);
+        await expect(governor).to.equal(governorSet);
+        await expect(guardian).to.equal(guardianSet);
+    });
+
+    it("Should revert if non executive calls onlyExecutive function", async function () {
```

Comment:
Another thing to check is that you can call onlyExecu methods by being an executive